rasa/utils/plotting.py

```diff
@@ -197,3 +199,75 @@ def plot_curve(
     plt.savefig(graph_path, format="pdf")
 
     logger.info(f"Comparison graph saved to '{graph_path}'.")
+
+
+def plot_intent_augmentation_summary(
```

Comment:
Is it possible to use the side-by-side histogram layout that `plot_histogram` uses for this as well? Of course we aren't plotting an histogram here but the layout is very clean IMO. What do you think?

Reply:
Do you have an example for such a lot at hand? Also, should we compare no augmentation vs augmentation, or random vs diverse augmentation, or both?

Reply:
It looks like this - 
![image](https://user-images.githubusercontent.com/8708249/105837644-00cd1a80-5ff5-11eb-90e1-8cd6a325aa93.png)
It's the confidence histogram that's plotted when you run `rasa test nlu` command

Reply:
Cool, thanks very much, I'll give it a go and see how it compares. I assume this is _instead_ of the existing plot we're currently producing, right?

Reply:
Right, plots are rotated now, looking like this.

![f1-score_changes](https://user-images.githubusercontent.com/480884/106128399-a7cdc580-615f-11eb-9746-03767da98138.png)

Reply:
@tttthomasssss Thanks! Do you think we should - 
1. Get rid of the grid lines, to make it look cleaner?
2. I had to think hard for a minute what `affected` and `total` were in the graph. I think that part of the information is clashing with what this graph intends to show. Maybe another thing to do could be we use a different colour for the labels of intent which were augmented, and all others are left black. Then the bars can stay blue and red indicating +ve or -ve change.

What do you think?

Reply:
ad 1. No strong opinions either way from my end, i.e. happy to remove them
ad 2. Thats a very good point. I think the current coding (augmented vs. affected vs. totals) would only make sense if developers had any direct influence of **which** intents are augmented, but they don't, so the current coding doesn't provide any useful information that they could act upon. Happy to use blue/red coding for +/-ve. I think, though, that using a different colour for the totals would be informative, though, because its contextualising the impact of augmentation and semantically the totals bars are somewhat different from the individual intents.

So, do you think colour coding should reflect +/- performance and e.g. hatching the totals makes sense?

Reply:
>  if developers had any direct influence of which intents are augmented, but they don't
But they still have influence on how many intents are augmented right?

> So, do you think colour coding should reflect +/- performance and e.g. hatching the totals makes sense?
Yes, colour coding should only reflect +/-ve performance IMO. How do you think you could hatch the totals? If you already have an idea in mind, would love to see how it looks.

Also, just noticed that the above plot does not show micro avg F1. It could be because the report contains accuracy and not micro F1. This should be factored in if not done already I guess.

Reply:
@dakshvar22 I've updated the plot presentation - there is no distinction now between augmented and affected as users don't have any influence over whats augmented, colours now reflect +ve/-ve and totals bars are hatched in addition to the colour coding. Let me know what you think.

![f1-score_changes](https://user-images.githubusercontent.com/480884/106728526-d559b980-660c-11eb-9a49-004f4a40a40d.png)

Reply:
Looks better I think, can we use the same color scheme/font scheme for +/-ve bars that intent histograms are using for correct/wrong predictions? Again the objective is just to stick to a standard design scheme as much as possible. Sorry for the nit-picks 🙈

Reply:
@dakshvar22 no bother :) - I've fiddled with the style and it looks more like the other one now!

Do you see any other loose ends that need tying?
![f1-score_changes](https://user-images.githubusercontent.com/480884/106767810-8d05c000-663b-11eb-99b8-5197cbd92df9.png)

Reply:
LGTM 💯